core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2580,7 +2580,8 @@ private[spark] class DAGScheduler(
     // if the cluster manager explicitly tells us that the entire worker was lost, then
     // we know to unregister shuffle output.  (Note that "worker" specifically refers to the process
     // from a Standalone cluster, where the shuffle service lives in the Worker.)
-    val fileLost = !sc.shuffleDriverComponents.supportsReliableStorage() &&
+    val fileLost = !(sc.shuffleDriverComponents.supportsReliableStorage() &&
+      sc.shuffleDriverComponents.supportsReliableStorage(execId)) &&
```

Comment:
What's the logic here? Is it `!(supportsReliableStorage() || supportsReliableStorage(execId))`?

Reply:
No. If we use `(supportsReliableStorage() || supportsReliableStorage(execId))`, newer implementation will always return true when use disaggregated shuffle cluster.

In old implementation `supportsReliableStorage(execId)` always get true since not implemented yet, thus the same behavior as just call `supportsReliableStorage()`
In newer implementation  `supportsReliableStorage()` should direct return  boolean, if it got true then `supportsReliableStorage(execId)` do real jugement

Reply:
Uh, the default value is true. Could we make it is false in default?

Reply:
> Uh, the default value is true. Could we make it is false in default?
Sorry I didn't see the other way to solve compatible issue, if make it false by default, shall you illustrate more

Reply:
If `supportsReliableStorage(execId)` returns false in default, so `!(supportsReliableStorage() || supportsReliableStorage(execId))` compatible with before.

Reply:
> If `supportsReliableStorage(execId)` returns false in default, so `!(supportsReliableStorage() || supportsReliableStorage(execId))` compatible with before.

what about the newer implementation

>  If we use (supportsReliableStorage() || supportsReliableStorage(execId)), newer implementation will always return true when use disaggregated shuffle cluster.

Reply:
When use disaggregated shuffle cluster, `supportsReliableStorage()` must be true since it's used in `ExecutorAllocationManager.validatings`

Reply:
`supportsReliableStorage()` returns true, means file not lost. right ?
`supportsReliableStorage(execId)` returns true, means file not lost too. right ?

Reply:
yes, When use disaggregated shuffle cluster `supportsReliableStorage(execId)` may return false since some local shuffle stages, while supportsReliableStorage() returns true

Reply:
I'm a bit confused.

Reply:
Ur right, it's really confused. How about now